### PR TITLE
[CALCITE-5446] Adding support for aliasing TIMESTAMP -> TIMESTAMP_WITH_LOCAL_TIMEZONE for BQ connections

### PR DESCRIPTION
**Pull request by olivrlee**
Adding in handling and type resolution for TIMESTAMP_WITH_LOCAL_TIMEZONE if the alias is set during connection

### CODE REVIEW

**Review by julianhyde — COMMENTED** [review]
I suggested some minor changes.

Can you revise the summary so that it describes the feature being added? It's nothing to do with JDBC, much more to do with https://issues.apache.org/jira/browse/CALCITE-5424.

The change was a bit confusing to review because the second commit negates the first. They could both have been squashed out.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlTimestampAddFunction.java [hunk 1]
```diff
@@ -109,6 +106,16 @@ public static RelDataType deduceType(RelDataTypeFactory typeFactory,
             || operandType2.isNullable());
   }
 
+  private static SqlTypeName sanitize(SqlTypeName typeName) {
+    switch (typeName) {
+    case TIME:
+    case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+      return typeName;
+    default: // usually DATE or TIMESTAMP
```
[the remark this answers is not in the document]
Comment (olivrlee) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlTimestampAddFunction.java:108]: I'll update wills comment.
It's saying that operations that involve hour/minute/second along with `DATE` or `TIMESTAMP` will result in `TIMESTAMP`

File: testkit/src/main/java/org/apache/calcite/test/ConnectionFactories.java [hunk 2]
```diff
@@ -162,6 +169,25 @@ private static class MapConnectionFactory implements ConnectionFactory {
     }
   }
 
+  /** Post-processor that adds a {@link Schema} and sets it as default. */
```
[the remark this answers is not in the document]
Comment (olivrlee) [reply testkit/src/main/java/org/apache/calcite/test/ConnectionFactories.java:172]: this seems to be copy pasted from `AddSchemaPostProcessor` above; I'll update.

File: babel/src/test/java/org/apache/calcite/test/BabelQuidemTest.java [hunk 3]
```diff
@@ -112,6 +113,12 @@ public static void main(String[] args) throws Exception {
               .with(CalciteConnectionProperty.CONFORMANCE,
                   SqlConformanceEnum.BABEL)
               .with(CalciteConnectionProperty.LENIENT_OPERATOR_LOOKUP, true)
+              .with(
+                  ConnectionFactories.addType("DATETIME", typeFactory ->
+                  typeFactory.createSqlType(SqlTypeName.TIMESTAMP)))
```
[the remark this answers is not in the document]
Comment (olivrlee) [reply babel/src/test/java/org/apache/calcite/test/BabelQuidemTest.java]: that causes tests to fail, what I have now is the automated fix after running  `./gradlew autostyleApply`

**Comment by olivrlee** [comment]
is there a way for me to be able to re-trigger tests? I see rerun available on my fork but not here

**Comment by asolimando** [comment]
@olivrlee, if I click on the "Details" link, then I can see the "re-run" option (both globally and individually per each task), but being a committer I don't know if I can see it for this reason.

Anyway, I have restarted the failed checks for you, hth.

**Pull request closed without merging**